fs/mount.c

```diff
@@ -23,6 +23,21 @@ void fs_register(const struct fs_ops *fs) {
     assert(!"reached filesystem limit");
 }
 
+char * get_filesystems(void) {
+    char *fs_list = calloc(MAX_FILESYSTEMS * 50, sizeof(char)); // Reasonable assumption?
```

Comment:
Thanks for sending this PR in! However, there's one problem here: your code allocates this buffer here, but it fails to free it. So each time the filesystems list is requested this leaks some memory. The buffer you're supposed to be writing to is already known to the procfs functions (it uses `proc_printf` which will do resizing as necessary) so I would suggest you actually use it to construct the string instead of creating a buffer here and limiting its size. What you may wish to do instead is return a list of filesystems instead of their string representations, and let the other file handle printing them. Once it's done, it'll free it.

Reply:
Got it.  I'll take a look.  Thank you for the feedback.